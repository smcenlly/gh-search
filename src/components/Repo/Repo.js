import React from "react";
import i18n from "i18next";
import styled from "styled-components";
import PropTypes from "prop-types";
import { Contributors } from "../Contributors";
import { colors } from "../../styles";

const RepoWrapper = styled.div`
    padding: 20px;
    & + & {
        border-top: 1px solid ${colors.separator};
    }
`;

class Repo extends React.PureComponent {
    static propTypes = {
        fullName: PropTypes.string,
        contributors: PropTypes.oneOfType([
            PropTypes.arrayOf(PropTypes.object),
            PropTypes.string
        ])
    };
    render() {
        let { className, fullName, contributors } = this.props;
        return (
            <RepoWrapper className={className}>
                <div>{i18n.t("repo.title", { fullName })}</div>
                <Contributors contributors={contributors} />
            </RepoWrapper>
        );
    }
}

export default Repo;
