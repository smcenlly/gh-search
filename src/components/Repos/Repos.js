import React from "react";
import i18n from "i18next";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Repo } from "../Repo";

const Wrapper = styled.div`
    overflow-y: auto;
`;

const NoReposFound = styled.div`
    padding: 20px;
`;

class Repos extends React.PureComponent {
    static propTypes = { repos: PropTypes.arrayOf(PropTypes.object) };
    static NoReposFound = NoReposFound;

    render() {
        let { className, repos } = this.props;
        return (
            <Wrapper className={className}>
                {repos.length ? (
                    repos.map(({ id, full_name, contributors }) => {
                        return (
                            <Repo
                                key={id}
                                fullName={full_name}
                                contributors={contributors}
                            />
                        );
                    })
                ) : (
                    <NoReposFound>{i18n.t("repos.notFound")}</NoReposFound>
                )}
            </Wrapper>
        );
    }
}

export default Repos;
