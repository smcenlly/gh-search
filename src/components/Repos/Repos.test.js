import React from "react";
import { shallow } from "enzyme";
import Repos from "./Repos";
import { Repo } from "../Repo";

describe("Repos", function() {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Repos repos={[]} />);
    });

    it("renders without crashing", () => {
        expect(wrapper).toExist();
    });

    it("matches snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe("when provided repos", () => {
        it("renders Repos with correct props", () => {
            const repos = [
                { id: 1, full_name: "RepoName1", contributors: [] },
                { id: 2, full_name: "RepoName2", contributors: [] }
            ];

            const repoProp = {
                fullName: repos[0].full_name,
                contributors: repos[0].contributors
            };

            wrapper.setProps({ repos });

            expect(wrapper).toContainMatchingElements(2, Repo);
            expect(wrapper.find(Repo).first()).toHaveProp(repoProp);
        });
    });

    describe("when not provided repos", () => {
        it("renders no repos found block", () => {
            expect(wrapper).toContainExactlyOneMatchingElement(
                Repos.NoReposFound
            );
        });
    });
});
