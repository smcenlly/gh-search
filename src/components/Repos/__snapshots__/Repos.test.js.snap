// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos matches snapshot 1`] = `
<div>
  <styled.div />
</div>
`;
